File: example/Storage/real.cpp
#include "cytnx.hpp"
#include <iostream>

using namespace cytnx;
int main() {
  /*
      Get the real part from a complex128 (ComplexDouble) Storage
  */
  Storage S1(10, Type.ComplexDouble);
  for (unsigned int i = 0; i < S1.size(); i++) {
    S1.at<cytnx_complex128>(i) = cytnx_complex128(i, i + 1);
  }
  std::cout << S1 << std::endl;

  Storage S1r = S1.real();
  std::cout << S1r << std::endl;

  /*
      Get the real part from a complex64 (ComplexFloat) Storage
  */
  Storage S2(10, Type.ComplexFloat);
  for (unsigned int i = 0; i < S1.size(); i++) {
    S2.at<cytnx_complex64>(i) = cytnx_complex64(i, i + 2);
  }
  std::cout << S2 << std::endl;

  Storage S2r = S2.real();
  std::cout << S2r << std::endl;

  return 0;
}

File: example/Storage/real.py
import sys
from pathlib import Path
home = str(Path.home())
sys.path.append(home + '/Cytnx_lib')
from cytnx import *


S1 = Storage(10,Type.ComplexDouble)

for i in range(10):
    S1[i] = i + 1j*(i+1)

S1r = S1.real()
print(S1)
print(S1r)

S2 = Storage(10,Type.ComplexFloat)

for i in range(10):
    S2[i] = i + 1j*(i+1)

S2r = S2.real()
print(S2)
print(S2r)
